R/weight.R

```diff
@@ -62,12 +124,115 @@ setMethod("weightVariables", "VariableCatalog", function (x) {
     ## Return the names/aliases
     if (length(vars)) {
         return(sort(vapply(index(vars), vget(namekey(x)), character(1),
-            USE.NAMES=FALSE)))
+            USE.NAMES = FALSE)))
     } else {
         return(c())
     }
 })
 
+#' @rdname weightVariables
+#' @export
+setMethod("weightVariables<-", "CrunchDataset", function (x, value) {
+    weightVariables(allVariables(x)) <- value
+    x <- refresh(x)
+    return(x)
+})
+
+#' @rdname weightVariables
+#' @export
+setMethod("weightVariables<-", "VariableCatalog", function (x, value) {
+    modifyWeightVariables(x, value, type = "append")
+})
+
+
+#' modifyWeightVariables
+#'
+#' Change which variables can be set as a dataset's weight.
+#'
+#' @param x a dataset or variable catalog
+#' @param vars Variables to add or remove  this can be a numeric Crunch variable,
+#' list of numeric Crunch variables or a character vector with the aliases of numeric Crunch variables.
+#' @param type a character string determining how the weightVariables
+#' will be modified:
+#' - `"append"` : add `vars` to the current weight variables
+#' - `"remove"` : remove `vars` from the current list of weight variables
+#' - `"replace"`: replace the current weight variables with `vars`
+#'
+#' @export
+modifyWeightVariables <- function (x, vars, type = "append") {
+    if (is.dataset(x)) {
+        x <- allVariables(x)
+    }
+    new <- old <- crGET(shojiURL(x, "orders", "weights"))
+    if (is.null(vars)) {
+        new$graph <- NULL
+        type <- "remove"
+    } else {
+        if (is.variable(vars) || (length(vars) == 1) & !is.character(vars)) {
+            vars <- list(vars)
+        }
+        if (is.character(vars)) {
+            ds <- loadDataset(datasetReference(self(x)))
+            vars <- lapply(vars, function (v) {
```

Comment:
Perhaps follow the pattern in `hideVariables`, which also takes aliases and gets their URLs: https://github.com/Crunch-io/rcrunch/blob/master/R/hide-variables.R#L40

The `[` method handles your validation of whether you're given valid aliases.

It takes a dataset as an argument, not a dataset or variable catalog, FWIW. Not sure you need to take anything other than dataset here either.

Reply:
I was just following the existing pattern for the `weightVariables()` getter, do we want to change everything to just take a dataset?

Reply:
The `weightVariables` getter doesn't return the dataset/catalog you gave it, it returns the aliases of the weight variables, so it's a little different. 

I'm fine with accepting both dataset and variable catalog if you can do it without undue complexity, but I don't view it as a requirement that the function must accept variable catalog as an argument.

Reply:
That's a good point. I guess it seemed like if `weightVariables(varcat)` worked then `weightVarialble(svarcat) <- variables` should also work.

Reply:
TBH I probably implemented the variable catalog method for `weightVariables` only because I had to get the catalog to get the weight order resource anyway, not because I expected/intended a user to do it.

Reply:
Great, I'll take out the VariableCatalog, that'll simplify things considerably